aspnetcore/fundamentals/host/generic-host.md

```diff
@@ -0,0 +1,406 @@
+
+**Key**: environment  
+**Type**: *string*  
+**Default**: Production  
+**Set using**: `UseEnvironment`  
+**Environment variable**: `ASPNETCORE_ENVIRONMENT`
+
+The environment can be set to any value. Framework-defined values include `Development`, `Staging`, and `Production`. Values aren't case sensitive. By default, the *Environment* is read from the `ASPNETCORE_ENVIRONMENT` environment variable. When using [Visual Studio](https://www.visualstudio.com/), environment variables may be set in the *launchSettings.json* file. For more information, see [Use multiple environments](xref:fundamentals/environments).
+
+[!code-csharp[](generic-host/samples-snapshot/2.x/GenericHostSample/Program.cs?name=snippet_UseEnvironment)]
+
+## ConfigureAppConfiguration
+
+App builder configuration is created by calling [ConfigureAppConfiguration(Action&lt;IConfigurationBuilder&gt;)](/dotnet/api/microsoft.extensions.hosting.ihostbuilder.configureappconfiguration) on the [IHostBuilder](/dotnet/api/microsoft.extensions.hosting.ihostbuilder) implementation. `ConfigureAppConfiguration` uses an [IConfigurationBuilder](/dotnet/api/microsoft.extensions.configuration.iconfigurationbuilder) to create an [IConfiguration](/dotnet/api/microsoft.extensions.configuration.iconfiguration) for the app. `ConfigureAppConfiguration` can be called multiple times with additive results. The app uses whichever option sets a value last. The configuration created by `ConfigureAppConfiguration` is available at [Configuration](/dotnet/api/microsoft.extensions.hosting.hostbuildercontext.configuration) for subsequent operations and in [Services](/dotnet/api/microsoft.extensions.hosting.ihost.services).
+
+*appsettings.json*:
+
+[!code-csharp[](generic-host/samples/2.x/GenericHostSample/appsettings.json)]
+
+*appsettings.Development.json*:
+
+[!code-csharp[](generic-host/samples/2.x/GenericHostSample/appsettings.Development.json)]
+
+*appsettings.Production.json*:
+
+[!code-csharp[](generic-host/samples/2.x/GenericHostSample/appsettings.Production.json)]
+
+Example app configuration using `ConfigureAppConfiguration`:
+
+[!code-csharp[](generic-host/samples-snapshot/2.x/GenericHostSample/Program.cs?name=snippet_ConfigureAppConfiguration)]
+
+> [!NOTE]
+> The [AddConfiguration](/dotnet/api/microsoft.extensions.configuration.chainedbuilderextensions.addconfiguration) extension method isn't currently capable of parsing a configuration section returned by [GetSection](/dotnet/api/microsoft.extensions.configuration.iconfiguration.getsection) (for example, `.AddConfiguration(Configuration.GetSection("section"))`. The `GetSection` method filters the configuration keys to the section requested but leaves the section name on the keys (for example, `section:Logging:LogLevel:Default`). The `AddConfiguration` method expects an exact match to configuration keys (for example, `Logging:LogLevel:Default`). The presence of the section name on the keys prevents the section's values from configuring the app.
```

Comment:
I used to show it in the sample ... [*until you killed it.*](https://github.com/aspnet/Docs/issues/5798#issuecomment-388525892) :trollface: :smile: lol

IIRC it was originally added to the hosting topic because devs were getting into trouble. Apparently, it's somewhat common to want to nest these config settings in the file and parse the sections out. I'll add back the link to the issue.

> This issue will be addressed in an upcoming release. For more information and workarounds, see \[Passing configuration section into WebHostBuilder.UseConfiguration uses full keys](https://github.com/aspnet/Hosting/issues/839).

Do you want to remove it (here only, or in both spots)?